docs/advanced/classes.rst

```diff
@@ -1261,3 +1261,37 @@ object, just like ``type(ob)`` in Python.
     Other types, like ``py::type::of<int>()``, do not work, see :ref:`type-conversions`.
 
 .. versionadded:: 2.6
+
+Custom type setup
+=================
+
+For advanced use cases, such as enabling garbage collection support, you may
+wish to directly manipulate the `PyHeapTypeObject` corresponding to a
+``py::class_`` definition.
+
+You can do that using ``py::custom_type_setup``:
+
+.. code-block:: cpp
+
+   struct OwnsPythonObjects {
+       py::object value = py::none();
+   };
+   py::class_<OwnsPythonObjects> cls(
+       m, "OwnsPythonObjects", py::custom_type_setup([](PyHeapTypeObject *heap_type) {
+           auto *type = &heap_type->ht_type;
+           type->tp_flags |= Py_TPFLAGS_HAVE_GC;
+           type->tp_traverse = +[](PyObject *self_base, visitproc visit, void *arg) {
```

Comment:
tbh
```
type->tp_traverse = +[](
----------_---------^
```
I don't know what the `+` does here. IIRC previously you had those also in test_custom_type_setup.cpp, but then you took them out there. I tried, you could also take them out here. Are they somehow important here? But not in the unit test?

Reply:
The `+` is applying unary `operator+` to the lambda, which causes it to be converted to a function pointer.  It is a somewhat common idiom.  However, it is true that it is not needed here, since that implicit conversion happens anyway, and in fact on Windows it causes problems which is why I took it out.

Reply:
This is now fixed.